module-sys/Service/Worker.cpp

```diff
@@ -265,9 +265,16 @@ namespace sys
     {
         assert(getState() == State::Running || getState() == State::Stopped);
         if (xSemaphoreTake(joinSemaphore, timeout) != pdTRUE) {
+            LOG_ERROR("Failed to take semaphore!");
             return false;
         }
-        while (eTaskGetState(taskHandle) != eDeleted) {}
+        const auto waitDeleteTaskTick = xTaskGetTickCount();
+        while (eTaskGetState(taskHandle) != eDeleted) {
+            if ((xTaskGetTickCount() - waitDeleteTaskTick) > timeout) {
+                LOG_ERROR("Timeout: Task state != eDeleted");
```

Comment:
Just a suggestion for this message: "Task waiting aborted (timeout). TaskState != eDeleted". It will be more descriptive.

Reply:
Done :)